proto/dydxprotocol/govplus/tx.proto

```diff
@@ -1,7 +1,58 @@
 syntax = "proto3";
 package dydxprotocol.govplus;
 
+import "cosmos_proto/cosmos.proto";
+import "cosmos/msg/v1/msg.proto";
+import "gogoproto/gogo.proto";
+import "amino/amino.proto";
+
 option go_package = "github.com/dydxprotocol/v4-chain/protocol/x/govplus/types";
 
 // Msg defines the Msg service.
-service Msg {}
+service Msg {
+  // SlashValidator is exposed to allow slashing of a misbehaving validator via
+  // governance.
+  rpc SlashValidator(MsgSlashValidator) returns (MsgSlashValidatorResponse);
+}
+
+// MsgSlashValidator is the Msg/SlashValidator request type.
+message MsgSlashValidator {
+  // The address that controls the module (likely the gov module account).
+  option (cosmos.msg.v1.signer) = "authority";
+  string authority = 1;
+
+  // Address of the validator to slash
+  string validator_address = 2;
+
+  // Colloquially, the height at which the validator is deemed to have
+  // misbehaved. In practice, this is the height used to determine the targets
+  // of the slash. For example, undelegating after this height will not escape
+  // slashing. This height should be set to a recent height at the time of the
+  // proposal to prevent delegators from undelegating during the vote period.
+  //
+  // NB: At the time this message is applied, this height must have occured
+  // equal to or less than an unbonding period in the past in order for the
+  // slash to be effective.
+  uint32 infraction_height = 3;
```

Comment:
Can we more explicitly document the suggested value for `infraction_height` in a proposal? Is it: `estimated_proposal_tally_height - unbonding_period < infraction_height < proposal_submission_height`, where the lowerbound is needed because `At the time this message is applied, this height must have occured equal to or less than an unbonding period in the past in order for the slash to be effective`.

Also in practice can we just suggest `infraction_height = proposal_submission_height`? Is there any use case where this should not be true?

Reply:
If someone knows they are getting slashed and undelegated before proposal.

Reply:
> If someone knows they are getting slashed and undelegated before proposal.

Makes sense. Could you add some documentation here for how `infraction_height` should be set, including an interval like suggested above?